### PR TITLE
refs #343: Fix walk and look points storage and retrieval

### LINKED ISSUE #343: Can't load saved games
### Bug description

Took from the Discord server by **DiamondWarrior**:

> When I load a saved game, it has a problem with loading the character's walk to point.

image

###  Steps to reproduce

1. Run a game.
2. Save the game.
3. Try to load the game.
4. The Editor will stop at the lines shown in the screenshot.

### Expected vs observed behavior

The game should load the saved data without issues.

### Environment information (please complete):

 - **OS type and version**: Windows 11
 - **Godot version**: 4.3
 - **Plugin version or branch**: 2.0.1

### PR DESCRIPTION
**Pull request by mapedorr**
Fixes #343 : The data should be stored using built-in types (int, float, String, and so on) in order to avoid hacks in Web games. A validation was added to assign the values for the `walk_to_point` and `look_at_point` based on the available data in the character's Dictionary.

### CODE REVIEW

File: addons/popochiu/engine/interfaces/i_room.gd
Comment (balloonpopper) [inline addons/popochiu/engine/interfaces/i_room.gd:250]: Should a check be done at this point to make sure that walk_to_point has the right format (i.e. "X,Y")? The script will die if walk_to_point contained a word for example, or didn't have a comma.

File: addons/popochiu/engine/interfaces/i_room.gd
Comment (balloonpopper) [inline addons/popochiu/engine/interfaces/i_room.gd:259]: As above, should the format of the variable be checked? Or are you happy to assume it doesn't have an invalid value?

**Comment by stickgrinder** [comment]
I would move this into a couple of functions, named something like `pack_vector_state` and `unpack_vector_state` (or anything that makes sense).

I agree with @balloonpopper about hardening the data with prior validation.

File: addons/popochiu/engine/interfaces/i_room.gd
Comment (mapedorr) [reply addons/popochiu/engine/interfaces/i_room.gd:250]: Could be, but this assumes we're dealing with the data as it was being stored in `popochiu_room_data.gd` prior to this fix. Anyway, I can add the validation; it never hurts.

File: addons/popochiu/engine/interfaces/i_room.gd
Comment (mapedorr) [reply addons/popochiu/engine/interfaces/i_room.gd:259]: Same reason of the previous one. I'll add the check.

**Comment by mapedorr** [comment]
@balloonpopper @stickgrinder I made an update using RegEx to validate the format of the stored **String**.

Regarding the creation of a function to pack those vectors (`PopochiuCharacter.walk_to_point` and `PopochiuCharacter.look_at_point`), I’m thinking about drafting a proposal to add a feature that automatically stores any **Vector2** property in objects saved as a JSON dictionary, with `x` and `y` keys and values.

Let me know what you think, and if it’s okay to merge this PR as it is (if approved), leaving the `pack_vector_state` function for the proposal I’m referring to.

**Comment by balloonpopper** [comment]
This looks complicated to me. Can we get away with just checking if it's the expected format as only Popochiu should be setting it and therefore the state should be known.
If you want to make it support all the use cases though (presumably you might be planning to use this in other places?) then it looks good.

I might be reading it wrongly (I'm not familiar enough with the codebase), but isn't the default case a dictionary? If so, I'd suggest putting the dictionary check before the regex.

**Comment by mapedorr** [comment]
When you suggested including a validation, I thought of a regular expression that could also cover us in case someone manually modifies the JSON file information. 

Right now, saved games can have a string `"Vector2(x, y)"` or `"(x, y)"`, which is why I also thought a **RegEx** was a good option. Once we merge the PR, it will start saving as a **Dictionary**. I don’t mind changing the order of the checks.

**Comment by stickgrinder** [comment]
@mapedorr this looks good to me. It solves the problem and is tidy.
I like your proposal for supporting Vectors state saving/loading. Can you open an "task" issue for that?

@balloonpopper do you think your open conversations could be resolved?

If you're just waiting for my validation, please go ahead and merge this fix!

**Review by stickgrinder — APPROVED** [review]

**Comment by balloonpopper** [comment]
I'm happy that my questions have been addressed

**Pull request merged**
